src/pages/issues/index.tsx

```diff
@@ -12,24 +12,22 @@ import { IssueItem } from '@/interfaces/issueItem.type';
 import { PullRequestAndIssueItem } from '@/interfaces/pullRequestIssueItem';
 
 type SearchFieldProps = {
-    searchText: string;
-    onSearchTextChanged: (event: ChangeEvent<HTMLInputElement>) => void;
-    onSearchTextSubmitted: () => void;
+    onSearchTextSubmitted: (searchString: string) => void;
     loading: boolean;
 };
 
-const SearchField = ({
-    searchText,
-    onSearchTextChanged,
-    onSearchTextSubmitted,
-    loading,
-}: SearchFieldProps) => {
+const SearchField = ({ onSearchTextSubmitted, loading }: SearchFieldProps) => {
+    const [searchText, setSearchText] = useState<string>('');
+    const onSearchTextChanged = (e: ChangeEvent<HTMLInputElement>) => {
+        setSearchText(e.target.value);
+    };
```

Comment:
- There is another improvement we can make here because the search action is only performed by clicking the `submit` button.
    - Instead of storing the search value in the state which currently has no benefit, the search query can be taken from the form using `FormData` and passed on to the `onSearchTextSubmitted` function.

Reply:
The button should be disabled or not is checked using the searchText. For this, we need the state which updates only when the onChange event is fired. 
IMO, this part is already optimized, keeping user experience in mind.

Let me know if you think otherwise. I will be happy discuss on this if you feel we should optimise it further.